tests/zfs-tests/tests/functional/cli_root/zpool_create/create-o_ashift.ksh

```diff
@@ -0,0 +1,161 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright 2016, loli10K. All rights reserved.
+#
+
+. $STF_SUITE/include/libtest.shlib
+. $STF_SUITE/tests/functional/cli_root/zpool_create/zpool_create.shlib
+
+#
+# DESCRIPTION:
+#	'zpool create -o ashift=<n> ...' should work with different ashift
+#	values.
+#
+# STRATEGY:
+#	1. Create various pools with different ashift values.
+#	2. Verify -o ashift=<n> works only with allowed values (9-16).
+#	   Also verify that the lowest number of uberblocks in a label is 16 and
+#	   smallest uberblock size is 8K even with higher ashift values.
+#
+
+verify_runnable "global"
+
+function cleanup
+{
+	poolexists $TESTPOOL && destroy_pool $TESTPOOL
+	log_must $RM $disk
+}
+
+#
+# FIXME: Ugly hack to force a $1 number of txg syncs.
```

Comment:
There's an existing ugly hack for this called `sync_pool` in `tests/zfs-tests/include/libtest.shlib`.  It's uses a similar trick but with `zpool scrub` instead.  You could use that until we do something better.  I like your version a lot except it has the side effect of leaving snapshots around.